Automattic/jetpack
# Carousel: reimplement without jQuery

Fixes #20036 
Fixes #11191  

Reimplement Carousel without `jQuery`, using only IE11-compatible vanilla ES5.

This is a big PR, but given the nature of the change, it would have been difficult to split it up into smaller chunks. Just like `jQuery`, it's pretty much an all-or-nothing affair.

Comparing the old and new code may be difficult, since the diffs are pretty much useless when just about every line of code has changed. As such, I'd recommend focusing on thorough testing instead.

I took an extremely slow and painful iterative approach, where I converted and tested one thing at a time through a lot of throwaway code, never allowing the carousel to be broken for more than a few minutes. I may have missed some bugs, but the result should be pretty close to the original in terms of functionality.

As for intentional changes, note that I removed a few sliding animations, since they tend to be slow, were difficult to implement in a performant way, and were only used for relatively minor things that don't benefit greatly from animation (such as displaying the status of a comment after submission). Fade animations were kept, but reimplemented in a more performant way.

#### Changes proposed in this Pull Request:
* Reimplement Carousel without `jQuery`

#### Jetpack product discussion
For a high-level, somewhat outdated plan, see p1HpG7-bfN-p2.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
This code touches all of the carousel, so it's difficult to provide any testing instructions. Please use the carousel as much as possible, trying all the different functionality, and see if anything breaks.

Things to watch out for:
- Does anything break if you take actions (like moving between slides, or opening and closing the carousel) too quickly?
- Does comment loading work correctly?
- Does comment posting work correctly?
- Does it still work in IE 11?
- Does it work in other browsers?
